src/components/EmbeddedDesignAttribute.tsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+
+export type LinkUpdatedCallback = (url: string) => Promise<string>
+
+interface AspectRatioProps {
+  width: number;
+  height: number;
+  children: any;
+}
+
+interface EmbeddedDesignInputProps {
+  product: string;
+  placeholder: string | undefined;
+  onInput: React.FormEventHandler<HTMLInputElement>;
+}
+
+export interface EmbeddedDesignAttributeProps {
+  identifier: string;
+  record: Aha.RecordUnion;
+  fields: { [index: string]: unknown };
+  product: string;
+  placeholder: string | undefined;
+  /** Optional async function that allows component to modify supplied user input before saving */
+  onLinkUpdated?: LinkUpdatedCallback
+}
+
+const AspectRatio = ({ width, height, children }: AspectRatioProps) => (
+  <div style={{ position: "relative", paddingTop: `${100 * height / width}%` }} >
+    <div style={{ position: "absolute", top: 0, left: 0, width: "100%", height: "100%" }}>
+      {children}
+    </div>
+  </div>
+)
+
+const EmbeddedDesignInput = ({ product, placeholder, onInput }: EmbeddedDesignInputProps) => {
+  return (
+    // @ts-expect-error: component definitions aren't available in this package. How do we install them?
```

Comment:
We can copy in the two .d.ts files that are in `.aha-cache/types` in any extension

Reply:
Copied them in. Is there an npm @types package I could install instead though?

Reply:
Though maybe I've copied them in too naively. `yarn build` is real mad now.

Reply:
no we don't provide a package for those